File: src/src/bmkim/level_2/큰수만들기.java
public class 큰수만들기 {
    public static void main(String[] args) {
        String number = "1924";
        int k = 2;
        System.out.println(solution(number, k));
    }

    public static String solution(String number, int k) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < number.length(); i++) {
            char currentChar = number.charAt(i);

            while (k > 0 && result.length() > 0 && result.charAt(result.length() - 1) < currentChar) {
                result.deleteCharAt(result.length() - 1);
                k--;
            }
            result.append(currentChar);
        }
        return result.toString();
    }
}

File: src/src/hvsun/level_2/큰수만들기.js
function solution(number, k) {
    const result = [];
    for (let i=0; i<number.length; i++) {
        // 1. 배열의 마지막 숫자가 현재 넣으려는 숫자보다 큰 지 검사
        // 2. 시도한 횟수가 제거할 수의 개수인 k를 초과하면 안됨
        while (result.length > 0 && result[result.length - 1] < number[i] && k > 0) {
            k--;
            result.pop();
        }
        result.push(number[i]);
    }
    result.splice(number.length - k, k);
    
    return result.join("");
}